ZenPacks/zenoss/LinuxMonitor/parsers/linux/cpu.py

```diff
@@ -48,17 +48,18 @@ def processResults(self, cmd, result):
             component = getattr(cmd, 'component', None)
 
             # Calculate the number of CPUs
-            cpuCount=0
+            cpuCount = 0
             single_cpu_line = ''
+
             for line in cmd.result.output.splitlines()[1:]:
-              words = line.split()[0:]
-              if 'cpu' in words[0]:
-                cpuCount+=1
-                if component:
-                  if component == line[3]:
-                    # If component, we matched the socket
-                    single_cpu_line = line
-                    break
+                # line e.g. 'cpu0 21306 149059 1785 18404 88442 0 117231 0 0 0'
+                words = line.split()[0:]
+                if 'cpu' in words[0]:
+                    cpuCount += 1
+                    if component and (component == line.split()[0][3:]):
```

Comment:
You already have words now, so use words[0] instead of the mysterious *line.split()[0][3:]*
The goal here is to make the code readble by a human.
Better yet, just set:

cpu = line.split()[0:] and use that throughout so we have a clue.

Reply:
This will also save one split()